xmppserver/src/main/java/org/jivesoftware/openfire/session/ServerSession.java

```diff
@@ -1,10 +1,50 @@
+/*
+ * Copyright (C) 2023 Ignite Realtime Foundation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package org.jivesoftware.openfire.session;
 
 public interface ServerSession extends Session {
+
+    enum AuthenticationMethod {
+        DIALBACK,
+        SASL_EXTERNAL,
+        OTHER
+    }
+
+    /**
+     * Obtain method that was used to authenticate this session. Null when the session is not authenticated.
+     *
+     * @return the method used for authentication (possibly null).
+     */
+    AuthenticationMethod getAuthenticationMethod();
```

Comment:
Perhaps
`Optional<AuthenticationMethod> getAuthenticationMethod();`
to force the check for absence?

Reply:
As I expected you to suggest this, initially implemented it as an Optional :).

I reverted that, as there was a silly amount of places where I had to 'get' the value without it doing a presence check.